app-react/src/main/kotlin/com/ustadmobile/view/HolidayCalendarDetailScreen.kt

```diff
@@ -0,0 +1,72 @@
+package com.ustadmobile.view
+
+import com.ustadmobile.core.generated.locale.MessageID
+import com.ustadmobile.core.hooks.useStringsXml
+import com.ustadmobile.core.impl.locale.StringsXml
+import com.ustadmobile.core.viewmodel.HolidayCalendarDetailUIState
+import com.ustadmobile.lib.db.entities.HolidayCalendar
+import com.ustadmobile.mui.components.UstadDetailField
+import mui.material.*
+import mui.system.responsive
+import react.FC
+import react.Props
+
+external interface HolidayCalendarDetailProps: Props{
+    var uiState: HolidayCalendarDetailUIState
+    var onItemClick: (HolidayCalendar) -> Unit
+}
+
+val HolidayCalendarDetailComponent2 = FC<HolidayCalendarDetailProps> { props ->
+
+    val strings: StringsXml = useStringsXml()
+
+    Container{
+        Stack{
+            spacing = responsive(2)
+
+            UstadDetailField{
+                labelText = strings[MessageID.name]
+                valueText = props.uiState.holidayCalendar?.umCalendarName ?: ""
+            }
+
+            List{
+                props.uiState.calendarList?.forEach { item ->
+                    ListItem {
+
+                        disablePadding = true
+
+                        ListItemButton {
+                            onClick = {
+                                props.onItemClick(item)
+                            }
+                            ListItemText {
```

Comment:
Please add secondary text for the holiday start and end time as per the Jetpack compose version. On React the function is useFormattedDate (instead of rememberFormattedDate ).

Reply:
useMemo is close - but should be done using useFormattedDate. useFormattedDate will format based on the language setting of the user (which might be different to the browser default).